### DISCUSSION TITLE
[Bug] useFormState errors produce stale data

### DISCUSSION THREAD

**Discussion by orenklein** [react-hook-form/react-hook-form#5097] [Q&A]
**Describe the bug**
In the following scenario the action flow is:
1. call setError manually
2. Clicking on submit -> clearErrors -> error seems to be cleared -> deffered handleSubmit is called -> error is back again

**To Reproduce**

1. Go to codesendbox
2. Error is manually triggered
3. click on submit -> clearErrors (error seems to be cleared)
4. handleSubmit is called after timeout -> error is back

In my specific scenario I needed to set error that is not associated with an input field, but the issue still occur if it is set on an input field.
1. uncomment ` const ERROR_PATH_TO_SUBSCRIBE = 'test.0.value';` still occur
2. uncomment in SomeErrorMessage `   const { formState:{errors} } = useFormContext(); ` - error is cleared even after handleSubmit.
3. uncomment ` handleSubmit(onSubmit)(e) ` instead of ` setTimeout... ` - still occur

**Codesandbox link (Required)**
https://codesandbox.io/s/quirky-visvesvaraya-f059d?file=/src/App.js

**Comment by bluebill1049** [comment]
hey @orenklein you should fire resubmit after the form is submitted at `useEffect`, take a look at the following example: 

https://codesandbox.io/s/brave-surf-gq20i?file=/src/App.js

**Reply by orenklein** [reply]
Thank you @bluebill1049 for the quick response (not sure if I need to tag you to trigger notification on comment reply)! 
I resubmitted with useeffect as you suggested, however it still doesn't work. 
In your example you don't call handleSubmit in form onSubmit, and it never gets inside `if(isSubmitted)`. Adding handleSubmit to form's onSubmit, issue the same behavior I described

**Reply by bluebill1049** [reply]
you are right. looks like a bug with `clearErrors` with empty argument.

**Reply by bluebill1049** [reply]
patching it now.

**Reply by orenklein** [reply]
@bluebill1049 FYA - even though it is mentioned in the release notes, the commit https://github.com/react-hook-form/react-hook-form/compare/v7.5.0...master is not inside the published release
